api/src/repositories/spatial-repository.ts

```diff
@@ -297,6 +297,18 @@ export class SpatialRepository extends BaseRepository {
     return response.rows;
   }
 
+  async deleteSpatialComponentsBySubmissionId(submission_id: number): Promise<ISubmissionSpatialComponent[]> {
+    const queryBuilder = getKnexQueryBuilder()
+      .del()
+      .from('submission_spatial_component')
+      .where({ submission_id })
+      .returning('*');
+
+    const response = await this.connection.knex<ISubmissionSpatialComponent>(queryBuilder);
+
+    return response?.rows || [];
```

Comment:
super nit: I believe rows is guaranteed to exist, and to be an array, so it should be safe to just return `response.rows`